src/restconfig/src/test/java/org/geoserver/rest/resources/ResourceTest.java

```diff
@@ -149,17 +149,24 @@ public void testResourceMetadataHTML() throws Exception {
     public void testResourceHeaders() throws Exception {
         MockHttpServletResponse response = getAsServletResponse("/rest/resource/mydir2/fake.png");
         Assert.assertEquals(FORMAT_HEADER.format(myRes.lastmodified()), response.getHeader("Last-Modified"));
-        Assert.assertEquals("/mydir2", response.getHeader("Resource-Parent"));
+        Assert.assertEquals("http://localhost:8080/geoserver/rest/resource/mydir2", 
+                response.getHeader("Resource-Parent"));
         Assert.assertEquals("resource", response.getHeader("Resource-Type"));
         Assert.assertEquals("image/png", response.getHeader("Content-Type"));
     }
 
     @Test
     public void testSpecialCharacterNames() throws Exception {
-        MockHttpServletResponse response = getAsServletResponse("/rest/resource/poëzie/café");
+        XMLUnit.setXpathNamespaceContext(NS_XML);
+        Document doc = getAsDOM("/rest/resource/po%c3%abzie?format=xml");
+        XMLAssert.assertXpathEvaluatesTo("http://localhost:8080/geoserver/rest/resource/po%c3%abzie/caf%c3%a9", 
+                "/ResourceDirectory/children/child/atom:link/@href", doc);
+
+        MockHttpServletResponse response = getAsServletResponse("/rest/resource/po%c3%abzie/caf%c3%a9");
```

Comment:
You could put "?format=xml" on this request to ensure that it is removed in Resource-Parent.